### ISSUE TITLE
gpt-4o model

### ISSUE THREAD

**Issue by xiejie2438** [openai/openai-openapi#274]
The model I passed in using the open AI API is gpt-4o, but I saw in the bill that my API key has the number of calls to other models. Does the gpt-4o model automatically convert to other models?

**Comment by benadamdev** [comment]
No, it doesn't. I think your API key was breached. You need to disable it and create a new one. If you're exposing the key inside a client (e.g iOS or Android app), you have to think of using a backend proxy.

**Comment by xiejie2438** [comment]
> No, it doesn't. I think your API key was breached. You need to disable it and create a new one. If you're exposing the key inside a client (e.g iOS or Android app), you have to think of using a backend proxy.

I have replaced the key, but there will still be calls to other models
![image](https://github.com/openai/openai-openapi/assets/38717856/6b09ee89-2287-4540-ad76-db9ba7ebc052)

**Comment by benadamdev** [comment]
It was breached again. Every network request you make to OpenAI is sniffable. Your only chance at fighting this is building your own backend or trying some service like https://www.aiproxy.pro 
I haven't tried them myself but I built my own proxy. Good luck

**Comment by xiejie2438** [comment]
Our requests to open AI are all initiated by backend services, not by clients such as browsers.

**Comment by benadamdev** [comment]
Someone must be using that key. I don't think there is any other explanation.

**Comment by montesclarosglennbenedict** [comment]
As @benadamdev said, the API key was likely compromised. Even if the requests are initiated by backend services, if the key is stored or transmitted insecurely it can be intercepted.

Have you considered internal misuse of the API key? Perhaps even backend services being compromised using whatever third-party packages you may be using. 

Monitor the usage logs, check the security of the backend services, and like @benadamdev use a proxy service. Wishing you luck with this one!

**Comment by mdegans** [comment]
> Every network request you make to OpenAI is sniffable.

This is not true. If your client is using TLS the key is not sniffable. So far as I know all generators use this by default.

> https://www.aiproxy.pro

Passing your API key through some random third party is an *awful* idea.

**Comment by StephenHodgson** [comment]
> Passing your API key through some random third party is an awful idea.

Agreed, that is why I have a [self hosted proxy service in my C# generated client](https://github.com/RageAgainstThePixel/OpenAI-DotNet/blob/main/OpenAI-DotNet-Proxy/Readme.md).

**Comment by jbeckwith-oai** [comment]
Thanks for the report. The `gpt-4o` alias resolving to the dated `gpt-4o-2024-05-13` snapshot is expected. A normal request to `gpt-4o` does not by itself explain usage attributed to unrelated model families or endpoints.

This repository tracks problems with the machine-readable OpenAPI specification. Please investigate the usage by time range, project, and API key in the Usage dashboard or Usage API. If a key may have been compromised, revoke it, and contact [OpenAI Support](https://help.openai.com/en/articles/6614161-how-can-i-contact-support) for account-specific investigation.

Closing as out of scope for this repository.
